### DISCUSSION TITLE
Netbox suddenly don't find my Custom Scripts

### DISCUSSION THREAD

**Discussion by ziggekatten** [netbox-community/netbox#7241] [General]
I'm at loss here.

Running my dev-environment (2.11.12) on Ubuntu for quite some time now, but suddenly Netbox does not find my scripts.

It picks up changes of SCRIPTS_ROOT correctly, but not the scripts. Tried to use different path and copy the files, at no avail.

Baffled, as this have been just working. Where should I start my troubleshooting?

**Comment by candlerb** [comment]
Have you checked logs when netbox restarts, or when you try to list scripts? (`journalctl -eu netbox`)

Are the permissions on your scripts directory (and outer directories) correct so that the netbox user can open them?  Check with:

```
sudo -u netbox ls -l /opt/netbox/netbox/scripts
```

Do you still have `/opt/netbox/netbox/scripts/__init__.py` ? If not, create it as a zero byte file.  You can also try deleting the `__pycache__` directory if it exists - it will be recreated automatically.

Maybe there's something invalid in one of the scripts which is killing it, although it shouldn't. You could try moving out the scripts and putting them back one by one.

When you say "It picks up changes of SCRIPTS_ROOT correctly, but not the scripts": what do you see which tells you that it's picking up a change to `SCRIPTS_ROOT` correctly?

**Reply by ziggekatten** [reply]
When going to Other->Scripts, i get the standard info message like:
![image](https://user-images.githubusercontent.com/55629049/132837831-4598c491-37b2-4fa2-b2de-94e4dc59b4ad.png)
If I change config, it's reflected.

But the issue seems to be sudo -u netbox ls -l /opt/netbox/netbox/scripts/
`ls: cannot access '/opt/netbox/netbox/scripts/': Permission denied`

Which is really weird.  I have to dig a little bit more, as there are soft links involved here

**Reply by ziggekatten** [reply]
Seems like latest version of vscode with remote development plugin messes with links/permissions somehow. 

Removed an recreated link and it works until switching branch and it breaks again. Problem fits timewise. Been working forever and I updated vscode yesterday.
